backend/controllers/appointmentDetails.js

```diff
@@ -13,7 +13,14 @@ module.exports = ({ prisonApi, whereaboutsApi, appointmentDetailsService }) => a
 
   const { additionalDetails, basicDetails, prepostData, recurringDetails, timeDetails } = appointmentViewModel
 
+  const userRoles = await oauthApi.userRoles(res.locals)
+
+  const canDeleteAppointment =
+    userRoles &&
+    userRoles.some(role => role.roleCode === 'ACTIVITY_HUB' || role.roleCode === 'DELETE_A_PRISONERS_APPOINTMENT')
+
   return res.render('appointmentDetails', {
+    appointmentConfirmDeletionLink: canDeleteAppointment ? `/appointment-details/${id}/confirm-deletion` : undefined,
```

Comment:
In case you didn't know, you can achieve the same thing by doing;  `canDeleteAppointment && `/appointment-details/${id}/confirm-deletion``